packages/fxa-auth-server/lib/routes/subscriptions/support.ts

```diff
@@ -0,0 +1,152 @@
+        const { uid, email } = await handleAuth(db, request.auth, true);
+        const { location } = request.app.geo;
+        await customs.check(request, email, 'supportRequest');
+
+        const {
+          productName,
+          topic,
+          app,
+          subject: payloadSubject,
+          message,
+        } = request.payload as Record<string, any>;
+        let subject = productName;
+        if (payloadSubject) {
+          subject = subject.concat(': ', payloadSubject);
+        }
+
+        const {
+          productNameFieldId,
+          locationCityFieldId,
+          locationStateFieldId,
+          locationCountryFieldId,
+          topicFieldId,
+          appFieldId,
+        } = config.zendesk;
+
+        const zendeskReq: zendesk.Requests.CreatePayload = {
+          request: {
+            comment: { body: message },
+            subject,
+            requester: {
+              email,
+              name: email,
+            },
+            custom_fields: [
+              { id: productNameFieldId, value: productName },
+              { id: topicFieldId, value: topic },
+              { id: appFieldId, value: app },
+              { id: locationCityFieldId, value: location.city },
+              { id: locationStateFieldId, value: location.state },
+              { id: locationCountryFieldId, value: location.country },
+            ],
+          },
+        };
+
+        let operation = 'createRequest';
+        try {
+          // Note that this awkward TypeScript conversion exists because the
+          // typings for this client fail to accomodate using the newer Promise
+          // return type.
+          const createRequest = ((await zendeskClient.requests.create(
+            zendeskReq
+          )) as unknown) as zendesk.Requests.ResponseModel;
+
+          const zenUid = createRequest.requester_id;
+
+          // 3 retries spread out over ~5 seconds
+          const retryOptions = {
+            retries: 3,
+            minTimeout: 500,
+            factor: 1.66,
+          };
+
+          // Ensure that the user has the appropriate custom fields
+          // We use retries here as they're more important for linking the
+          // Zendesk user to the fxa uid.
+          operation = 'showUser';
+          const showRequest = await pRetry(async () => {
+            return ((await zendeskClient.users.show(
+              zenUid
+            )) as unknown) as zendesk.Users.ResponseModel;
+          }, retryOptions);
+          const userFields = showRequest.user_fields as
+            | null
+            | undefined
+            | Record<string, any>;
+          if (!userFields?.user_id || !showRequest.locale) {
+            operation = 'updateUser';
+            await pRetry(async () => {
+              return await zendeskClient.users.update(zenUid, {
+                user: {
+                  user_fields: { user_id: uid },
+                  locale: request.app.locale,
```

Comment:
Maybe you already know and decided that it's not a problem.  Just in case though: `request.app.locale` is accept-language filtered through our list of supported languages for the frontends and emails.  'de' and 'fr' are on that list, so it's not a problem right now.

The idea was to add the language tag as a custom field.  Maybe this is better for routing the support ticket?  I don't know enough abut Zendesk.  But we should check with JR.  Where does this locale end up in Zendesk?

Reply:
In a locale field to mark the users locale. I did talk to @dannycoates to verify that this value is correct.

Reply:
Note: checked and resolved with JR in Slack